File: src/main/java/com/kisman/cc/features/module/render/shader/shaders/OutlineShader.java
package com.kisman.cc.features.module.render.shader.shaders;

import com.kisman.cc.features.module.render.shader.FramebufferShader;
import org.lwjgl.opengl.GL20;

public final class OutlineShader extends FramebufferShader {
    public static final OutlineShader OUTLINE_SHADER = new OutlineShader();
    public float radius, quality, red,  green, blue, alpha, rainbowSpeed, rainbowStrength, saturation;

    public OutlineShader() {
        super("outline_custom.frag");
    }

    @Override
    public void setupUniforms() {
        setupUniform("texelSize");
        setupUniform("color");
        setupUniform("divider");
        setupUniform("radius");
        setupUniform("maxSample");
        setupUniform("rainbowStrength");
        setupUniform("rainbowSpeed");
        setupUniform("saturation");
    }

    @Override
    public void updateUniforms() {
        GL20.glUniform2f(getUniform("texelSize"), 1F / mc.displayWidth * (radius * quality), 1F / mc.displayHeight * (radius * quality));
        GL20.glUniform4f(getUniform("color"), red, green, blue, alpha);
        GL20.glUniform1f(getUniform("radius"), radius);
        GL20.glUniform2f(getUniform("rainbowStrength"), rainbowStrength, rainbowStrength);
        GL20.glUniform1f(getUniform("rainbowSpeed"), rainbowSpeed);
        GL20.glUniform1f(getUniform("saturation"), saturation);
    }
}

File: src/main/java/com/kisman/cc/features/module/Debug/futureshader/shaders/outline/OutlineShader.kt
package com.kisman.cc.features.module.Debug.futureshader.shaders.outline

import com.kisman.cc.features.module.Debug.futureshader.FramebufferShader
import org.lwjgl.opengl.GL20

/**
 * @author _kisman_
 * @since 11:31 of 10.06.2022
 */
object OutlineShader : FramebufferShader("outline", "outline") {
    var opacity = 1f
    var width = 1f
    var quality = 1f

    override fun setupUniforms() {
        setupUniform("texture")
        setupUniform("texelSize")
        setupUniform("opacityModifier")
    }

    override fun updateUniforms() {
        GL20.glUniform1i(getUniform("texture"), 0)
        GL20.glUniform2f(getUniform("texelSize"), 1 / mc.displayWidth * width * quality, 1 / mc.displayHeight * width * quality)
        GL20.glUniform1f(getUniform("opacityModifier"), opacity)
    }
}